core/state-machine/src/overlayed_changes.rs

```diff
@@ -134,25 +240,27 @@ impl OverlayedChanges {
 		if self.committed.is_empty() {
 			::std::mem::swap(&mut self.prospective, &mut self.committed);
 		} else {
-			for (key, val) in self.prospective.drain() {
-				let entry = self.committed.entry(key).or_default();
+			for (key, val) in self.prospective.top.drain() {
+				let entry = self.committed.top.entry(key).or_default();
 				entry.value = val.value;
 
 				if let Some(prospective_extrinsics) = val.extrinsics {
 					entry.extrinsics.get_or_insert_with(Default::default)
 						.extend(prospective_extrinsics);
 				}
 			}
-		}
-	}
+			for (storage_key, map) in self.prospective.children.drain() {
+				let entry = self.committed.children.entry(storage_key).or_default();
```

Comment:
Can be written as:
`self.committed.children.entry(storage_key).or_default().extend(map.1.iter())`

Reply:
Needed to write it as

```
let entry = self.committed.children.entry(storage_key).or_default();
entry.1.extend(map.1.iter().map(|(k, v)| (k.clone(), v.clone())));
```

Not sure if there's better ways.

Reply:
Does `map.1.iter().cloned()` maybe work?

Reply:
No it doesn't:

```
type mismatch resolving `<std::collections::hash_map::Iter<'_, std::vec::Vec<u8>, std::option::Option<std::vec::Vec<u8>>> as std::iter::Iterator>::Item == &_`
```

I think we already have some instance of tuple reference clone over map like this (for example, when computing delta for storage root in `Ext`). It just look like `cloned` doesn't work well for those cases.

Reply:
Okay :)